src/fdmdv2_main.cpp

```diff
@@ -3764,50 +3938,48 @@ void txRxProcessing()
                 codec2_fifo_write(cbData->outfifo1, outsound_card, nout);
             }
             else {
-                nout = resample(cbData->outsrc2, outsound_card, outfreedv, g_soundCard1SampleRate, FS, N48, N8);
+                nout = resample(cbData->outsrc2, outsound_card, outfreedv, g_soundCard1SampleRate, freedv_get_speech_sample_rate(g_pfreedv), N48, g_bufferSize);
                 codec2_fifo_write(cbData->outfifo1, outsound_card, nout);
             }
         }
         else {
-            nout = resample(cbData->outsrc2, outsound_card, outfreedv, g_soundCard2SampleRate, FS, N48, N8);
+            nout = resample(cbData->outsrc2, outsound_card, outfreedv, g_soundCard2SampleRate, freedv_get_speech_sample_rate(g_pfreedv), N48, g_bufferSize);
             codec2_fifo_write(cbData->outfifo2, outsound_card, nout);
         }
     }
- 
+
     //
     //  TX side processing --------------------------------------------
     //
 
     if ((g_mode != -1) && ((g_nSoundCards == 2) && ((g_half_duplex && g_tx) || !g_half_duplex))) {
         int ret;
 
-	if (g_dump_fifo_state) {
-	  // just dump outfifo1 state atm as that is causing problems with 700D
-	  // If this drops to zero we have a problem as we will run out of output samples
-	  // to send to the sound driver via PortAudio
-	  fprintf(stderr, "%6d", codec2_fifo_used(cbData->outfifo1));
-	}
-
         // This while loop locks the modulator to the sample rate of
         // sound card 1.  We want to make sure that modulator samples
         // are uninterrupted by differences in sample rate between
         // this sound card and sound card 2.
 
-        // Run this while loop as soon as we have enough room for one
-        // frame of modem samples.  Aim is to keep outfifo1 nice and
-        // full so we don't have any gaps ix tx signal.
+        // Run code inside this while loop as soon as we have enough
+        // room for one frame of modem samples.  Aim is to keep
+        // outfifo1 nice and full so we don't have any gaps ix tx
+        // signal.
 
         unsigned int nsam_one_modem_frame = g_soundCard2SampleRate * freedv_get_n_nom_modem_samples(g_pfreedv)/freedv_samplerate;
+        
+ 	if (g_dump_fifo_state) {
+	  // If this drops to zero we have a problem as we will run out of output samples
+	  // to send to the sound driver via PortAudio
+	  fprintf(stderr, "outfifo1 used: %6d free: %6d nsam_one_modem_frame: %d\n",
+                  codec2_fifo_used(cbData->outfifo1), codec2_fifo_free(cbData->outfifo1), nsam_one_modem_frame);
+	}
 
+        int nsam_in_48 = g_soundCard2SampleRate * freedv_get_n_speech_samples(g_pfreedv)/freedv_get_speech_sample_rate(g_pfreedv);
+        assert(nsam_in_48 < 10*N48);
         while((unsigned)codec2_fifo_free(cbData->outfifo1) >= nsam_one_modem_frame) {
 
```

Comment:
@ac0zj I think this should be ```while((unsigned)codec2_fifo_free(cbData->outfifo1) >= nsam_one_modem_frame)``` to ensure output fifo stays full?

Perhaps we can now try smaller FIFO sizes?